### PR TITLE
Allow users to change and reset their own passwords

### LINKED ISSUE #1047: Allow users to reset their own passwords
On the login page, there should be an "I forgot my password" button which lets a user request a link to reset their password.

There is already [a view](https://github.com/opensafely-core/opencodelists/blob/main/opencodelists/views/user_set_password.py) that lets a user set their password.  [This was added](https://github.com/opensafely-core/opencodelists/commit/1d617342962a618711735f90892a2b32e8b6b17a#diff-399d753ed281242a40bc79c6b334a8a659494b692462012dce893cd2c7688eea) before users could sign up themselves (the workflow was that I'd create an account, send them this link, and they'd set their own password) and this might be repurposable for this.

### PR DESCRIPTION
**Pull request by rebkwok**
Fixes #1047 

Uses django's auth urls/views, so just requires adding relevant templates, and setting up email.

### CODE REVIEW

File: opencodelists/settings.py [hunk 1]
```diff
@@ -259,3 +260,16 @@
...
+ANYMAIL = {
+    "MAILGUN_API_KEY": env.str("MAILGUN_API_KEY", default=None),
+    "MAILGUN_API_URL": "https://api.eu.mailgun.net/v3",
+    "MAILGUN_SENDER_DOMAIN": "mg.opencodelists.opensafely.org",
+}
+EMAIL_BACKEND = env.str(
+    "EMAIL_BACKEND", default="django.core.mail.backends.console.EmailBackend"
+)
+DEFAULT_FROM_EMAIL = "you@example.com"
+SERVER_EMAIL = "your-server@example.com"
```
Comment (ghickman) [inline opencodelists/settings.py:275]: We should probably set these to something 😁

**Review by ghickman — CHANGES_REQUESTED** [review]
Couple of config values need fixing.

You've added an HTML email template, do you know how Django handles text-only content for this email?

Really happy to see this being added!

File: opencodelists/settings.py (hunk 1 above)
Comment (rebkwok) [reply opencodelists/settings.py:275]: We should probably also change them on job-server, where I copied the settings from :smile:

**Comment by rebkwok** [comment]
> You've added an HTML email template, do you know how Django handles text-only content for this email?

It's an html template to override the one that the default view uses, but I think it just sends it as text
https://github.com/django/django/blob/main/django/contrib/auth/views.py#L222

File: opencodelists/settings.py (hunk 1 above)
Comment (rebkwok) [reply opencodelists/settings.py:275]: Are these sensible values?
```
DEFAULT_FROM_EMAIL = "no-reply@opencodelists.org"
SERVER_EMAIL = "tech@opensafely.org"
```

Comment (ghickman) [reply opencodelists/settings.py:275]: I think those make sense 👍

**Comment by ghickman** [comment]
Oh yes, I see, there's an `html_email_template_name` too, perfect 👍

**Review by ghickman — APPROVED** [review]

**Pull request merged**